# Data Inventory and Initial Assessment

## Purpose

Before performing any cleaning or analysis, I want to understand the structure, contents, and quality of the available Fitbit datasets.

The Bellabeat case study provides two Fitbit data exports covering different time periods. Before selecting a dataset for analysis, I will inventory the available files, inspect their structure, evaluate data completeness, and determine whether the datasets should be analyzed separately or combined.

The goal of this notebook is to document the dataset selection process and establish a foundation for the remainder of the project.

In [1]:
import pandas as pd
import numpy as np
from pathlib import Path

In [3]:
from pathlib import Path

export1 = Path(r"C:\Users\Weekseey\Documents\Data Projects\bellabeat-analysis\data\raw\fitbit_export_3.12.16-4.11.16")
export2 = Path(r"C:\Users\Weekseey\Documents\Data Projects\bellabeat-analysis\data\raw\fitbit_export_4.12.16-5.12.16")

files1 = sorted([file.name for file in export1.glob("*.csv")])
files2 = sorted([file.name for file in export2.glob("*.csv")])

print("Export 1 Files:", len(files1))
print("Export 2 Files:", len(files2))

Export 1 Files: 11
Export 2 Files: 18


In [4]:
print("Export 1")
for file in files1:
    print(file)

print("\n" + "="*50 + "\n")

print("Export 2")
for file in files2:
    print(file)

Export 1
dailyActivity_merged.csv
heartrate_seconds_merged.csv
hourlyCalories_merged.csv
hourlyIntensities_merged.csv
hourlySteps_merged.csv
minuteCaloriesNarrow_merged.csv
minuteIntensitiesNarrow_merged.csv
minuteMETsNarrow_merged.csv
minuteSleep_merged.csv
minuteStepsNarrow_merged.csv
weightLogInfo_merged.csv


Export 2
dailyActivity_merged.csv
dailyCalories_merged.csv
dailyIntensities_merged.csv
dailySteps_merged.csv
heartrate_seconds_merged.csv
hourlyCalories_merged.csv
hourlyIntensities_merged.csv
hourlySteps_merged.csv
minuteCaloriesNarrow_merged.csv
minuteCaloriesWide_merged.csv
minuteIntensitiesNarrow_merged.csv
minuteIntensitiesWide_merged.csv
minuteMETsNarrow_merged.csv
minuteSleep_merged.csv
minuteStepsNarrow_merged.csv
minuteStepsWide_merged.csv
sleepDay_merged.csv
weightLogInfo_merged.csv


## User Overlap Investigation

Before selecting a final dataset, I want to determine whether the two Fitbit exports contain the same participants. If user IDs overlap and the date ranges are consecutive, the datasets may be combined to increase the analysis period and sample size.

In [5]:
df1 = pd.read_csv(export1 / "dailyActivity_merged.csv")
df2 = pd.read_csv(export2 / "dailyActivity_merged.csv")

users1 = set(df1["Id"].unique())
users2 = set(df2["Id"].unique())

print("Users in Export 1:", len(users1))
print("Users in Export 2:", len(users2))

overlap = users1.intersection(users2)

print("Shared Users:", len(overlap))

Users in Export 1: 35
Users in Export 2: 33
Shared Users: 33


In [6]:
print("\nExport 1 IDs:")
print(sorted(users1))

print("\nExport 2 IDs:")
print(sorted(users2))


Export 1 IDs:
[1503960366, 1624580081, 1644430081, 1844505072, 1927972279, 2022484408, 2026352035, 2320127002, 2347167796, 2873212765, 2891001357, 3372868164, 3977333714, 4020332650, 4057192912, 4319703577, 4388161847, 4445114986, 4558609924, 4702921684, 5553957443, 5577150313, 6117666160, 6290855005, 6391747486, 6775888955, 6962181067, 7007744171, 7086361926, 8053475328, 8253242879, 8378563200, 8583815059, 8792009665, 8877689391]

Export 2 IDs:
[1503960366, 1624580081, 1644430081, 1844505072, 1927972279, 2022484408, 2026352035, 2320127002, 2347167796, 2873212765, 3372868164, 3977333714, 4020332650, 4057192912, 4319703577, 4388161847, 4445114986, 4558609924, 4702921684, 5553957443, 5577150313, 6117666160, 6290855005, 6775888955, 6962181067, 7007744171, 7086361926, 8053475328, 8253242879, 8378563200, 8583815059, 8792009665, 8877689391]


## Preliminary Findings: User Overlap

The initial comparison of participant IDs revealed substantial overlap between the two Fitbit exports.

### Key Findings

- Export 1 contains data from 35 unique users.
- Export 2 contains data from 33 unique users.
- All 33 users present in Export 2 also appear in Export 1.
- Two users (2891001357 and 6391747486) appear only in Export 1 and do not continue into Export 2.

### Initial Interpretation

The results suggest that the two exports likely represent consecutive collection periods from the same Fitbit study rather than two independent datasets.

Because nearly all participants appear in both exports, combining the datasets may provide a longer observation period and a more complete view of user behavior. However, before making a final dataset selection decision, the collection dates must be examined to determine whether the exports are consecutive and whether any overlapping records exist.

In [7]:
print("Export 1 Date Range")
print("Start:", df1["ActivityDate"].min())
print("End:", df1["ActivityDate"].max())

print("\nExport 2 Date Range")
print("Start:", df2["ActivityDate"].min())
print("End:", df2["ActivityDate"].max())

Export 1 Date Range
Start: 3/12/2016
End: 4/9/2016

Export 2 Date Range
Start: 4/12/2016
End: 5/9/2016


## Preliminary Findings: Date Range Comparison

The date ranges of the two Fitbit exports were compared to determine whether the datasets could be combined for analysis.

### Key Findings

- Export 1 contains records from March 12, 2016 through April 9, 2016.
- Export 2 contains records from April 12, 2016 through May 9, 2016.
- The exports do not overlap.
- A two-day gap exists between the datasets (April 10-11, 2016).

### Interpretation

The results indicate that the two exports represent consecutive phases of the same Fitbit study. Although a small two-day gap exists between the collection periods, the participant overlap and continuity of dates suggest that combining the datasets will provide a more complete view of user behavior than analyzing either export independently.

### Preliminary Decision

The two exports will be merged during the data preparation phase. The two-day gap will be documented as a limitation but is not expected to materially impact the analysis.

In [8]:
print("Export 1 Rows:", len(df1))
print("Export 2 Rows:", len(df2))

Export 1 Rows: 457
Export 2 Rows: 940


In [9]:
df1.head()

,Id,ActivityDate,TotalSteps,TotalDistance,TrackerDistance,LoggedActivitiesDistance,VeryActiveDistance,ModeratelyActiveDistance,LightActiveDistance,SedentaryActiveDistance,VeryActiveMinutes,FairlyActiveMinutes,LightlyActiveMinutes,SedentaryMinutes,Calories
0,1503960366,3/25/2016,11004,7.11,7.11,0.0,2.57,0.46,4.07,0.0,33,12,205,804,1819
1,1503960366,3/26/2016,17609,11.55,11.55,0.0,6.92,0.73,3.91,0.0,89,17,274,588,2154
2,1503960366,3/27/2016,12736,8.53,8.53,0.0,4.66,0.16,3.71,0.0,56,5,268,605,1944
3,1503960366,3/28/2016,13231,8.93,8.93,0.0,3.19,0.79,4.95,0.0,39,20,224,1080,1932
4,1503960366,3/29/2016,12041,7.85,7.85,0.0,2.16,1.09,4.61,0.0,28,28,243,763,1886


In [10]:
df2.head()

,Id,ActivityDate,TotalSteps,TotalDistance,TrackerDistance,LoggedActivitiesDistance,VeryActiveDistance,ModeratelyActiveDistance,LightActiveDistance,SedentaryActiveDistance,VeryActiveMinutes,FairlyActiveMinutes,LightlyActiveMinutes,SedentaryMinutes,Calories
0,1503960366,4/12/2016,13162,8.50,8.50,0.0,1.88,0.55,6.06,0.0,25,13,328,728,1985
1,1503960366,4/13/2016,10735,6.97,6.97,0.0,1.57,0.69,4.71,0.0,21,19,217,776,1797
2,1503960366,4/14/2016,10460,6.74,6.74,0.0,2.44,0.40,3.91,0.0,30,11,181,1218,1776
3,1503960366,4/15/2016,9762,6.28,6.28,0.0,2.14,1.26,2.83,0.0,29,34,209,726,1745
4,1503960366,4/16/2016,12669,8.16,8.16,0.0,2.71,0.41,5.04,0.0,36,10,221,773,1863


## Dataset Selection Decision

The two Fitbit exports were evaluated to determine whether they should be analyzed independently or merged into a single dataset.

### Evidence Supporting a Merge

- Export 1 contains 457 records across 35 users.
- Export 2 contains 940 records across 33 users.
- All users present in Export 2 are also present in Export 1.
- Only two users appear exclusively in Export 1.
- The datasets cover consecutive collection periods with only a two-day gap.
- Both datasets share the same schema, column structure, and data types.

### Decision

The two exports will be merged into a single master dataset for analysis.

### Rationale

Combining the datasets increases the observation period from approximately one month to nearly two months while maintaining participant continuity. This provides a more complete representation of user behavior and strengthens the reliability of subsequent analyses.

### Limitation

A two-day gap exists between April 9, 2016 and April 12, 2016. This gap will be documented but is not expected to materially affect the analysis.